### PR TITLE
Pass all vars

### PR DESCRIPTION
**Pull request by simPod**
Built on top of https://github.com/grafana/simple-json-datasource/pull/95

Should resolve https://github.com/grafana/simple-json-datasource/issues/78

### CODE REVIEW

**Comment by andig** [comment]
ping @maintainers is there any chance to get this PR and the corresponding #95 merged? 

My use case is a timeseries data API for a monitoring application (https://volkszaehler.org) that allows returning current or consumption data (latter grouped by hour, day etc). Currently, there is not way to pass the grouping parameters to the data source except for adhoc filters which is kinda ugly.

@simPod would it somehow be possible to allow variables (or another mechanism) per panel instead of per dashboard? Given use case above I might want to have monthly consumption plus current day timeseries which would require separate parameters.

**Comment by simPod** [comment]
@andig just yesterday, I finished a feature that allows to pass additional json data for each target. Didn't want to spam PRs here though as my other 4 are still open and 2 are hanging here for a month almost with no feedback from repo maintainers...

**Comment by andig** [comment]
If you dont mind publishing the plugin on your own I‘d be happy to use yours while we‘re waiting for merge. Might need a nudge in the forums after a month?

**Comment by simPod** [comment]
@daniellee would any maintainer have time to check these? :)

**Comment by andig** [comment]
ping @daniellee any chance looking into this PR and the ones it depends on? It would be really nice to get these squared away in order to continue refining https://github.com/grafana/simple-json-datasource/pull/104 which is high value for me as it greatly enhances the datasource's flexibility.

**Comment by andig** [comment]
This PR simply adds the user-defined variables (tag key/values) to the query in 
https://github.com/grafana/simple-json-datasource/pull/100/files#diff-3c16d1e97d76c1b44c74a650c7dbc5b2R33.

We have meanwhile moved this into https://github.com/simPod/grafana-json-datasource. If this is just an example then the PR can be closed.

**Comment by daniellee** [comment]
Oh ok, I see what this does now. It adds existing template variables to the scoped vars. I suppose it could be useful if you are not using the template variable in the target and want to pass it to the backend. It's not how it is usually done with datasources so feels like a bit of a workaround. Most datasources do the templating on the frontend when building the query. The interpolated query is then sent to the backend. So I'm not sure if this PR is useful to that many people.

It does not close #78 as far as I can see. Or am I missing something?

**Comment by avgdev** [comment]
> ...  Most datasources do the templating on the frontend when building the query. The interpolated query is then sent to the backend. So I'm not sure if this PR is useful to that many people.
> 
This is PR is very useful to a lot of people as there is no way to specify the template variable in the API call  unlike other datasource where the template variable is written into the query before it is interpolated. 

If I have 50 targets, how do i tell the API to return data for the specific target I am selecting? Is there a way to do that?

**Comment by propertone** [comment]
+1

File: src/datasource.js [hunk 1]
```diff
@@ -30,6 +30,18 @@ export class GenericDatasource {
       query.adhocFilters = [];
     }
 
+    const index = _.isUndefined(this.templateSrv.index) ? {} : this.templateSrv.index;
```
Comment (torkelo) [inline src/datasource.js]: this is not allowed, not sure what your trying to do here and what the point of doing this is

**Comment by torkelo** [comment]
I do not understand this PR, is it trying to interpolate ad hoc filters? but this is done in way that is not allowed (different from all other data sources) and accessing internal templateSrv stuff.

**Comment by simPod** [comment]
I'll look at the feedback ASAP (just after I finish one feature on my project)

Just quickly, this adds variables that were originally missing in the request. Those defined in the dashboard in Variables section.

There's a bug I know about that needs to be fixed, now it works in certain cases only. What would be an alternative way if templateSrv is private? From where it is possible to fetch dashboard variables and its current values?

File: src/datasource.js (hunk 1 above)
Comment (simPod) [reply src/datasource.js]: This is fetching all set variables so they can be sent to API endpoint

**Pull request closed without merging**